### PR TITLE
snapclient: Fixed incompatible types when listing sound cards and added --host support

### PR DESCRIPTION
**Pull request by EvTheFuture**
The listing of sound cards failed due to the previous method used returned bytes instead of str.

### CODE REVIEW

**Comment by heitbaum** [comment]
Could you please update the commit comment as per house style.

Could you please also update the changelog and the package makefile so that the addon is picked up as an update.

**Comment by heitbaum** [comment]
As a secondary note: there is request to enable http json as an option in snapserver (as you are in that area - might be something you could look at 😀🙏)

https://forum.libreelec.tv/thread/23750-snapserver-enable-http-json-rpc-http-post-and-websockets/?postID=153424#post153424

**Comment by EvTheFuture** [comment]
> Could you please update the commit comment as per house style.
>
I'll do that.

> Could you please also update the changelog and the package makefile so that the addon PKG_REV is picked up as an update.

I will create one more PR today that I need in order for the snapclient to work for me. Namely an extra option to specify the IP Adress/hostname to the server. 

Can I just update the Makefile in that commit to avoid two new consecutive versions?

**Comment by EvTheFuture** [comment]
> As a secondary note: there is request to enable http json as configurable option in snapserver (as you are in that area - might be something you could look at 😀🙏)
> 
> https://forum.libreelec.tv/thread/23750-snapserver-enable-http-json-rpc-http-post-and-websockets/?postID=153424#post153424

Yes, I can fix this after I fix option on the client to specify where the server can be found.as.mentioned above.

**Comment by heitbaum** [comment]
> > Could you please update the commit comment as per house style.
> 
> I'll do that.
> 
> > Could you please also update the changelog and the package makefile so that the addon PKG_REV is picked up as an update.
> 
> I will create one more PR today that I need in order for the snapclient to work for me. Namely an extra option to specify the IP Adress/hostname to the server.
> 
> Can I just update the Makefile in that PR to avoid two new consecutive versions?

Yup. Thought that might be approach. Create a seperate commit for each of the changes, but will only need the 1 add on rev update.

**Comment by EvTheFuture** [comment]
> Yup. Thought that might be approach. Create a seperate commit for each of the changes, but will only need the 1 add on rev update.

Should I create a separate PR for the new command line option or shall I include it as a commit in this PR?

Please not I fixed an issue (most likely introduced in this PR, that the last card listed was not included). Shall i include that in on commit or as two separate commits as now?

**Comment by EvTheFuture** [comment]
After a fresh reboot, there seem to be something else that doesn't work as expected on my system. I'll double check to see why.

**Comment by EvTheFuture** [comment]
> After a fresh reboot, there seem to be something else that doesn't work as expected on my system. I'll double check to see why.

Yes, my bad and it's fixed now.

**Comment by heitbaum** [comment]
> > Yup. Thought that might be approach. Create a seperate commit for each of the changes, but will only need the 1 add on rev update.
> 
> Should I create a separate PR for the new command line option or shall I include it as a commit in this PR (I interpreted your answer as the latter)?
> 
> **Please note**: I fixed an issue (most likely introduced in this PR, that the last card listed was not included). Shall i include that as an --amend commit or as a separate commit as I have done now?
> 
> Sorry for the questions, this is the first time contributing to this project.

Correct you will want to amend your commit to be a “complete fix”. While developing you can have them as seperate commits, but you will need to squash them as the single fix come PR time.

**Comment by EvTheFuture** [comment]
> +1 [0ac7cf4](https://github.com/LibreELEC/LibreELEC.tv/commit/0ac7cf42948418d84f6ba201eef4e7a0467a4907) and [394542e](https://github.com/LibreELEC/LibreELEC.tv/commit/394542e83b3390476785ceb59b560f2350bc3d2d) belong in the same commit.

Thank you for the clarification. I did a fixup of the 2nd commit and have pushed. There are now only two commits. One for the sound card list fix and one for the --host parameter.

File: packages/addons/service/snapclient/source/addon.py [hunk 1]
```diff
@@ -9,9 +9,10 @@
 SNAPCLIENT = os.path.join(
     xbmcaddon.Addon().getAddonInfo('path'), 'bin', 'snapclient')
 
+line = ''
 card = ''
 cards = []
-lines = subprocess.check_output([SNAPCLIENT, '--list']).splitlines()
+lines = subprocess.getoutput(SNAPCLIENT + ' --list').splitlines()
```
Comment (mglae) [inline packages/addons/service/snapclient/source/addon.py:15]:
```suggestion
lines = subprocess.run([SNAPCLIENT, '--list'], capture_output=True, text=True).stdout.splitlines()
```
Python3 docu recommends using subprocess.run() when possible. Untested.

**Comment by EvTheFuture** [comment]
Cleaned up the commits

**Comment by mglae** [comment]
Do you continue with:

> As a secondary note: there is request to enable http json as configurable option in snapserver [...]

Or is this good to be merged?

**Comment by EvTheFuture** [comment]
> Do you continue with:
> 
> > As a secondary note: there is request to enable http json as configurable option in snapserver [...]
> 
Yes but as a separate PR. I haven't got time yet, but plan to do it shortly.

> Or is this good to be merged?

This PR is good to be merged from my side.

**Comment by EvTheFuture** [comment]
> Do you continue with:
> 
> > As a secondary note: there is request to enable http json as configurable option in snapserver [...]
> 
Ibe been working with the server a bit before, so if i haven't missed anything in the request, it's fairly easy to fix.

I've created a än addon for Home Assistant for the snapcast server [here](https://github.com/EvTheFuture/hassio-addon-snapcast)

**Review by mglae — APPROVED** [review]

**Pull request merged**